### ISSUE TITLE
Updated whisper doesn't use the settings of new components

### ISSUE THREAD

**Issue by hans-hsieh-asus** [open-olive/loop-development-kit#217]
I ran the code below.
```javascript
const myWhisper = await whisper.create({
  label: 'hello',
  components: [
    {
      type: whisper.WhisperComponentType.Checkbox,
      value: false,
      label: 'hello',
      onChange: () => {},
    },
  ],
  onClose: () => {},
});
await delay(5000);
myWhisper.update({
  label: 'hello2',
  components: [
    {
      type: whisper.WhisperComponentType.Checkbox,
      value: false,
      label: 'hello2',
      onChange: () => {},
    },
  ],
}, (err) => {
  console.error(err);
});
```
And I check the checkbox immediately.
Wait for 5 seconds. The text is changed from `hello` to `hello2`, but the new checkbox is also checked.
The expected behavior is that the new checkbox should be unchecked.
The same behavior also happens for the `open` status of a `CollapseBox`.

Another related use case we have is that we have a whisper with 10 checkboxes. And we put the last 5 checkboxes in a `CollapseBox`. When a user checks one checkbox, the whisper is updated to add a `Message`. If the user expanded the `CollapseBox` before checking the checkbox, we want the `CollapseBox` to remain open after the update.  The current behavior is: For the 1st update, the `open` flag of the `CollapseBox` will be respected. However, if the `CollapseBox` was open before the update, and we set `open` to true for the update, there will be an animation to expand the `CollapseBox`. The user experience isn't very good. For the 2nd and later update, the `open` flag of the `CollapseBox` won't be respected. The `CollapseBox` always stays in the same state as before the update.

p.s. This is the same issue I reported in Slack. I just create this one to help to track.

**Comment by michaellatman** [comment]
Related to #215

**Comment by epiccoleman** [comment]
We've made some changes to how components update in our next release, which should solve this problem with CollapseBox. WhisperComponents will now have an optional 'key' property which can be used to set an identifier on the Component, allowing its state to persist properly across updates. 

It's probably worth noting that the current design of the Checkbox means that Loop authors need to set their checked/unchecked state manually. The forthcoming changes to Whisper updates won't affect this.

**Comment by hans-hsieh-asus** [comment]
@epiccoleman Thanks. Manually setting the checkbox state is ok. And we are already doing so. :)

**Comment by CandiceCLB** [comment]
This issue has been resolved with the latest LDK release, V3.2.0.
